query-node/mappings/giza/storage/index.ts

```diff
@@ -0,0 +1,636 @@
+  const operatorStatus = new StorageBucketOperatorStatusActive()
+  operatorStatus.workerId = workerId.toNumber()
+  storageBucket.operatorStatus = operatorStatus
+  await store.save<StorageBucket>(storageBucket)
+}
+
+export async function storage_StorageBucketInvitationCancelled({
+  event,
+  store,
+}: EventContext & StoreContext): Promise<void> {
+  const [bucketId] = new Storage.StorageBucketInvitationCancelledEvent(event).params
+  const storageBucket = await getById(store, StorageBucket, bucketId.toString())
+  const operatorStatus = new StorageBucketOperatorStatusMissing()
+  storageBucket.operatorStatus = operatorStatus
+  await store.save<StorageBucket>(storageBucket)
+}
+
+export async function storage_StorageBucketOperatorInvited({
+  event,
+  store,
+}: EventContext & StoreContext): Promise<void> {
+  const [bucketId, workerId] = new Storage.StorageBucketOperatorInvitedEvent(event).params
+  const storageBucket = await getById(store, StorageBucket, bucketId.toString())
+  const operatorStatus = new StorageBucketOperatorStatusInvited()
+  operatorStatus.workerId = workerId.toNumber()
+  storageBucket.operatorStatus = operatorStatus
+  await store.save<StorageBucket>(storageBucket)
+}
+
+export async function storage_StorageBucketOperatorRemoved({
+  event,
+  store,
+}: EventContext & StoreContext): Promise<void> {
+  const [bucketId] = new Storage.StorageBucketInvitationCancelledEvent(event).params
+  const storageBucket = await getById(store, StorageBucket, bucketId.toString())
+  const operatorStatus = new StorageBucketOperatorStatusMissing()
+  storageBucket.operatorStatus = operatorStatus
+  await store.save<StorageBucket>(storageBucket)
+}
+
+export async function storage_StorageBucketsUpdatedForBag({
+  event,
+  store,
+}: EventContext & StoreContext): Promise<void> {
+  const [bagId, addedBucketsIds, removedBucketsIds] = new Storage.StorageBucketsUpdatedForBagEvent(event).params
+  // Get or create bag
+  const storageBag = await getBag(store, bagId)
+  const assignmentsToRemove = await store.getMany(StorageBagStorageAssignment, {
+    where: {
+      storageBag,
+      storageBucket: { id: In(Array.from(removedBucketsIds).map((bucketId) => bucketId.toString())) },
+    },
+  })
+  const assignmentsToAdd = Array.from(addedBucketsIds).map(
+    (bucketId) =>
+      new StorageBagStorageAssignment({
+        id: `${storageBag.id}-${bucketId.toString()}`,
+        storageBag,
+        storageBucket: new StorageBucket({ id: bucketId.toString() }),
+      })
+  )
+  await Promise.all(assignmentsToRemove.map((a) => store.remove<StorageBagStorageAssignment>(a)))
+  await Promise.all(assignmentsToAdd.map((a) => store.save<StorageBagStorageAssignment>(a)))
+}
+
+export async function storage_StorageBucketDeleted({ event, store }: EventContext & StoreContext): Promise<void> {
+  const [bucketId] = new Storage.StorageBucketDeletedEvent(event).params
+  // TODO: Delete or just change status?
```

Comment:
Please, resolve TODOs

Reply:
Done in https://github.com/Joystream/joystream/pull/2515/commits/3f04581173a720ac3a15a10c673d19875e0b4961